src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -645,33 +657,37 @@ public static void WriteProjectFile(
             writer.WriteLine();
         }
 
-        // Kept in sync with the default `dotnet new console` project file (enforced by `DotnetProjectAddTests.SameAsTemplate`).
-        writer.WriteLine($"""
-              <PropertyGroup>
-                <OutputType>Exe</OutputType>
-                <TargetFramework>net10.0</TargetFramework>
-                <ImplicitUsings>enable</ImplicitUsings>
-                <Nullable>enable</Nullable>
-                <PublishAot>true</PublishAot>
-              </PropertyGroup>
-            """);
+        // Write default properties unless they are overridden by custom directives.
+        var propertyDirectiveNames = propertyDirectives.Select(d => d.Name).ToHashSet(StringComparer.OrdinalIgnoreCase);
+        if (!s_defaultProperties.Keys.All(propertyDirectiveNames.Contains))
```

Comment:
Honestly, this feels pretty complicated. It took me a few seconds to even understand what this line was doing. Why do we need to have many different property groups? Why not just open a property group, write the default ones (unless the user specified them), then EnableDefaultCompileItems, then any user-defined properties, all in one group?

Reply:
At the very least, I'd remove this check, and do something more like this, for clarity:

```cs
bool openedDefaultGroup = false;
foreach (var (name, value) in s_defaultProperties)
{
    if (!propertyDirectiveNames.Contains(name))
    {
        if (!openedDefaultGroup)
        {
            openedDefaultGroup = true;
            writer.WriteLine("  <PropertyGroup>");
        }

        writer.WriteLine($"""
            <{name}>{EscapeValue(value)}</{name}>
        """);
    }
}
if (openedDefaultGroup)
{
    writer.WriteLine("  </PropertyGroup>");
    writer.WriteLine();
}
```

Reply:
> Why do we need to have many different property groups? Why not just open a property group, write the default ones (unless the user specified them), then EnableDefaultCompileItems, then any user-defined properties, all in one group?

Sounds good to me, I will simplify like that.